cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationHelper.java

```diff
@@ -0,0 +1,101 @@
+
+/**
+ * This class contains methods for getting current configuration from the kafka brokers asynchronously.
+ */
+public class KafkaBrokerConfigurationHelper {
+
+    private final AdminClientProvider adminClientProvider;
+    protected final SecretOperator secretOperations;
+    private static final Logger log = LogManager.getLogger(KafkaBrokerConfigurationHelper.class);
+
+    public KafkaBrokerConfigurationHelper(AdminClientProvider adminClientProvider, SecretOperator secretOperations) {
+        this.adminClientProvider = adminClientProvider;
+        this.secretOperations = secretOperations;
+    }
+
+    public Future<Admin> adminClient(StatefulSet sts, int podId) {
+        Promise<Admin> acPromise = Promise.promise();
+        String cluster = sts.getMetadata().getLabels().get(Labels.STRIMZI_CLUSTER_LABEL);
+        String namespace = sts.getMetadata().getNamespace();
+        Future<Secret> clusterCaKeySecretFuture = secretOperations.getAsync(
+                namespace, KafkaResources.clusterCaCertificateSecretName(cluster));
+        Future<Secret> coKeySecretFuture = secretOperations.getAsync(
+                namespace, ClusterOperator.secretName(cluster));
+        String hostname = KafkaCluster.podDnsName(namespace, cluster, KafkaCluster.kafkaPodName(cluster, podId)) + ":" + KafkaCluster.REPLICATION_PORT;
+
+        return CompositeFuture.join(clusterCaKeySecretFuture, coKeySecretFuture).compose(compositeFuture -> {
+            Secret clusterCaKeySecret = compositeFuture.resultAt(0);
+            if (clusterCaKeySecret == null) {
+                return Future.failedFuture(Util.missingSecretException(namespace, KafkaCluster.clusterCaKeySecretName(cluster)));
+            }
+            Secret coKeySecret = compositeFuture.resultAt(1);
+            if (coKeySecret == null) {
+                return Future.failedFuture(Util.missingSecretException(namespace, ClusterOperator.secretName(cluster)));
+            }
+
+            Admin ac;
+            try {
+                log.info("about to create an admin client");
+                ac = adminClientProvider.createAdminClient(hostname, clusterCaKeySecret, coKeySecret, "cluster-operator");
+                acPromise.complete(ac);
+            } catch (RuntimeException e) {
+                log.warn("Failed to create Admin Client. {}", e);
+                acPromise.complete(null);
+            }
+            return acPromise.future();
+        });
+    }
+
+    public Future<Map<ConfigResource, Config>> getCurrentConfig(int podId, Admin ac) {
+        Promise<Map<ConfigResource, Config>> futRes = Promise.promise();
+        ConfigResource resource = new ConfigResource(ConfigResource.Type.BROKER, String.valueOf(podId));
+        DescribeConfigsResult configs = ac.describeConfigs(Collections.singletonList(resource));
+        Map<ConfigResource, Config> config = new HashMap<>();
+        try {
+            config = configs.all().get(2000, TimeUnit.MILLISECONDS);
```

Comment:
How do you know 2 seconds will be enough? Don't we need some BackOff here? Thsi seems quite agressive.